src/Sulu/Bundle/ContentBundle/Resources/public/js/components/roles/ckeditor/main.js

```diff
@@ -0,0 +1,190 @@
+
+            this.sandbox.start([
+                {
+                    name: 'matrix@husky',
+                    options: {
+                        el: '#matrix-container',
+                        captions: {
+                            all: this.translations.all,
+                            none: this.translations.none,
+                            type: this.translations.type,
+                            horizontal: this.translations.horizontal,
+                            vertical: values.vertical
+                        },
+                        values: values,
+                        data: data
+                    }
+                }
+            ]);
+        },
+
+        prepareMatrixValues: function() {
+            var vertical = [],
+                horizontal = [];
+
+            for (var key in this.toolbar) {
+                if (!this.toolbar.hasOwnProperty(key)) {
+                    continue;
+                }
+
+                vertical.push(key);
+                horizontal.push(_.map(this.toolbar[key], function(button) {
+                    return {value: button, icon: this.sandbox.ckeditor.getIcon(button), title: button};
+                }.bind(this)));
+            }
+
+            return {vertical: vertical, horizontal: horizontal};
+        },
+
+        prepareMatrixData: function() {
+            var data = [];
+            for (var section in this.toolbar) {
+                data.push(
+                    _.map(this.toolbar[section], function(item) {
+                        return _.contains(this.data.toolbar[section], item);
+                    }.bind(this))
+                );
+            }
+
+            return data;
+        },
+
+        changeData: function(data) {
+            if (typeof(data.value) !== 'string') {
+                this.sandbox.dom.each(data.value, function(key, value) {
+                    this.changeData({section: data.section, value: value.value, activated: data.activated});
+                }.bind(this));
+
+                return;
+            }
+
+            if (!!data.activated) {
+                if (_.contains(this.data.toolbar[data.section], data.value)) {
+                    return;
+                }
+
+                if (!this.data.toolbar[data.section]) {
+                    this.data.toolbar[data.section] = [];
+                }
+
+                this.data.toolbar[data.section].push(data.value);
+
+                return;
+            }
+
+            var index = this.data.toolbar[data.section].indexOf(data.value);
+            if (index > -1) {
+                this.data.toolbar[data.section].splice(index, 1);
+            }
+        },
+
+        save: function(action) {
+            this.sandbox.emit('sulu.header.toolbar.item.loading', 'save');
+
+            $.ajax('/admin/api/roles/' + this.data.id + '/settings/ckeditor-toolbar', {
```

Comment:
Should we really put 'ckeditor' in our routes. Can we make it more generic with 'text-editor'?